### PR TITLE
Reduce scale factor and val loss across ranks for DDP

### LINKED ISSUE #1458: Reduce scale factor for DDP in generative model
**Is your feature request related to a problem? Please describe.**
Currently, scale factor is computed for each rank for DDP in generative model. 
It should be reduced across ranks to use the same scale factor. 
It does not cause a problem for performance for now because scale factors are all close to one in this example.
But in the future it might cause issue.

Same issue for validation loss. 

**Describe the solution you'd like**
Reduce scale factors and validation loss across ranks. 

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Can-Zhao**
Fixes #1458 .

### Description
Reduce scale factor and val loss across ranks for DDP

### Checks

- [ ] Avoid including large-size files in the PR.
- [ ] Clean up long text outputs from code cells in the notebook.
- [ ] For security purposes, please check the contents and remove any sensitive info such as user names and private key.
- [ ] Ensure (1) hyperlinks and markdown anchors are working (2) use relative paths for tutorial repo files (3) put figure and graphs in the `./figure` folder
- [ ] Notebook runs automatically `./runner.sh -t <path to .ipynb file>`

### CODE REVIEW

**Review by guopengf — APPROVED** [review]
LGTM
